File: VB/WindowsApplication1/Custom Ribbon Control/MyRibbonViewInfo.vb
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Ribbon.ViewInfo

Namespace WindowsApplication1
    Public Class MyRibbonViewInfo
        Inherits RibbonViewInfo

        Public Sub New(ByVal ribbon As RibbonControl)
            MyBase.New(ribbon)
        End Sub
        Protected Overrides Function CreatePanelInfo() As RibbonPanelViewInfo
            Return New MyRibbonPanelViewInfo(Me)
        End Function
    End Class
End Namespace

File: CS/WindowsApplication1/Custom Ribbon Control/MyRibbonViewInfo.cs
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.ViewInfo;

namespace WindowsApplication1 {
    public class MyRibbonViewInfo : RibbonViewInfo {
        public MyRibbonViewInfo(RibbonControl ribbon)
            : base(ribbon) {
        }
        protected override RibbonPanelViewInfo CreatePanelInfo() {
            return new MyRibbonPanelViewInfo(this);
        }
    }
}
